htlcswitch/link_test.go

```diff
@@ -975,6 +975,48 @@ func TestUpdateForwardingPolicy(t *testing.T) {
 	default:
 		t.Fatalf("expected FailFeeInsufficient instead got: %v", err)
 	}
+
+	// Reset the policy so we can then test updating the max HTLC policy.
+	n.secondBobChannelLink.UpdateForwardingPolicy(n.globalPolicy)
+
+	// As a sanity check, ensure the original payment now succeeds again.
+	_, err = makePayment(
+		n.aliceServer, n.carolServer, firstHop, hops, amountNoFee,
+		htlcAmt, htlcExpiry,
+	).Wait(30 * time.Second)
+	if err != nil {
+		t.Fatalf("unable to send payment: %v", err)
+	}
+
+	// Now we'll update Bob's policy to lower his max HTLC to an extent
+	// that'll cause him to reject the same HTLC that we just sent.
+	newPolicy = n.globalPolicy
+	newPolicy.MaxHTLC = amountNoFee - 1
+	n.secondBobChannelLink.UpdateForwardingPolicy(newPolicy)
+
+	// Next, we'll send the payment again, using the exact same per-hop
+	// payload for each node. This payment should fail as it won't factor
+	// in Bob's new max HTLC policy.
+	_, err = makePayment(
+		n.aliceServer, n.carolServer, firstHop, hops, amountNoFee,
+		htlcAmt, htlcExpiry,
+	).Wait(30 * time.Second)
+	if err == nil {
+		t.Fatalf("payment should've been rejected")
+	}
+
+	ferr, ok = err.(*ForwardingError)
+	if !ok {
+		t.Fatalf("expected a ForwardingError, instead got (%T): %v",
+			err, err)
+	}
+	switch ferr.FailureMessage.(type) {
+	case *lnwire.FailTemporaryChannelFailure:
```

Comment:
Is this the error we return? :thinking: For a policy problem, we'd rather have a different error, so we can retry with the new channel update if that was attached.

Reply:
TemporaryChannelFailure was [the only error option I saw](https://github.com/lightningnetwork/lightning-rfc/blob/master/04-onion-routing.md#failure-messages) that includes a `ChannelUpdate` and isn't inaccurate to the reason for failure. Lmk if I'm missing a more descriptive failure type option.

It seems like since the spec has an error for `htlc_minimum_msat` it should have an error for `htlc_maximum_msat` too :thinking:

Reply:
Without an `htlc_maximum_msat` error, you'd need to make an assumption on the sender's implementation and what triggers them to retry. For `lnd` senders, I'd say it is best to send `htlc_minimum_msat` also for exceeding the max. What we'll do is apply the channel update and retry. But maybe you want to hear other's opinions about this too.

Reply:
@wpaulino, thoughts?

Reply:
> For a policy problem, we'd rather have a different error, so we can retry with the new channel update if that was attached.

I think we can proceed with this as is for now. We already return the error in the master branch, the change being done here is for a test.

Reply:
Ok, good point. We don't make it worse here.